ibis/common/scope.py

```diff
@@ -0,0 +1,65 @@
+""" Module for scope
+    Scope is a dictionary mapping :class:`~ibis.expr.operations.Node`
+    subclass instances to concrete data, and the time context associate
+    with it(if any).
+    Scope is used in many backends to cache data for calculated ops.
+"""
+from typing import Optional
+
+from ibis.expr.typing import TimeContext
+from ibis.timecontext.util import TimeContextRelation, compare_timecontext
+
+
+def set_scope_item(op, result, timecontext: Optional[TimeContext]):
+    """make a scope item to be set in scope
```

Comment:
`set_scope_item` is perhaps not the right make for this function. I think this is more like "make_scope_item" or sth?

Reply:
Actually, no this is a complete scope dictionary.

Do you handle merging scopes that contains different timecontext for the same op?

Reply:
Yes it is handle in set method now. Refactor `make_scope_item` and `set_scope_item` to be separate functions